### PR TITLE
Update workspace role to allow get/list/create/delete secrets

### LINKED ISSUE #483: Add a role to be able to create, list and delete secrets by a namespace user
Need to add a role and bind it to a namespace user, to be able to create, list and delete secrets. See: https://github.com/eclipse-che/che-server/pull/44

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
This PR is an update to https://github.com/devfile/devworkspace-operator/pull/489. Instead of granting access to all secrets in the namespace, we scope the permission down to single secret named `workspace-credentials-secret`.

### What issues does this PR fix or reference?
Closes #483

### Is it tested? How?
Test that workspace SA can create/update/delete secret named `workspace-credentials-secret`. I haven't tested how the `list` permission interacts with a restricted role in this way (will do later)

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v7-devworkspaces-operator-e2e, v7-devworkspace-happy-path` to trigger)
    - [ ] `v7-devworkspaces-operator-e2e`: DevWorkspace e2e test
    - [ ] `v7-devworkspace-happy-path`: DevWorkspace e2e test

### CODE REVIEW

**Review by sleshchenko — APPROVED** [review]

File: controllers/workspace/provision/rbac.go [hunk 1]
```diff
@@ -56,6 +56,12 @@ func generateRBAC(namespace string) []runtime.Object {
 					APIGroups: []string{"apps", "extensions"},
 					Verbs:     []string{"get", "list", "watch"},
 				},
+				{
+					Resources:     []string{"secrets"},
+					APIGroups:     []string{""},
+					Verbs:         []string{"get", "list", "create", "delete"},
```
[the remark this answers is not in the document]
Comment (amisevsk) [reply controllers/workspace/provision/rbac.go:62]: Thanks for testing my laziness -- I'll remove :)

[the remark this answers is not in the document]
Comment (JPinkney) [reply controllers/workspace/provision/rbac.go:62]: @amisevsk I see that you removed list from the other places, should you remove it here as well?

[the remark this answers is not in the document]
Comment (amisevsk) [reply controllers/workspace/provision/rbac.go:62]: Yup... :facepalm: totally missed fixing it in the file where the comment was originally made.

**Review by JPinkney — APPROVED** [review]

**Comment by amisevsk** [comment]
Happy-path is failing with error

```
E0811 22:27:35.804519       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.Event: failed to list *v1.Event: events is forbidden: User "system:serviceaccount:devworkspace-controller:devworkspace-controller-serviceaccount" cannot list resource "events" in API group "" at the cluster scope
```

I've no idea what's going on there (works when testing locally in minikube). Merging this PR to get it off my backlog and the issue seems unrelated.

**Pull request merged**